pkg/kn/commands/service/create_test.go

```diff
@@ -618,6 +618,116 @@ func TestServiceCreateScaleWithMinScaleSet(t *testing.T) {
 
 }
 
+func TestServiceCreateScaleRange(t *testing.T) {
+	action, created, _, err := fakeServiceCreate([]string{
+		"service", "create", "foo", "--image", "gcr.io/foo/bar:baz",
+		"--scale", "1..5", "--no-wait"}, false)
+
+	if err != nil {
+		t.Fatal(err)
+	} else if !action.Matches("create", "services") {
```

Comment:
I would separate them as two different checks. No need to be if/else since either execution will stop and fail the test.

Reply:
I separated these, however the original blocks were copied from other tests and I was trying to keep it consistent. Many of the other code blocks in create_test and update_test use:

```
	if err != nil {
		t.Fatal(err)
	} else if !action.Matches("create", "services") {
		t.Fatalf("Bad action %v", action)
	}
```

Should I revert this back for consistency?

Reply:
if this is the case, then typically I would recommend to move the shared code pattern into a dedicated function for validation like in a `validateAction(t, action, err, []string{ "create", "services")` to guarantee that the same code is used everywhere.

However as this is not in scope for this pr I would recommend to do this in a separate pr and for now just keep the pattern so that it can be easily detected when doing the full refactoring.